### PR TITLE
named tensor support max pooling functions

### PR DESCRIPTION
**Pull request by glaringlee**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#31340 named tensor support max pooling functions**

Differential Revision: [D19131355](https://our.internmc.facebook.com/intern/diff/D19131355)

### CODE REVIEW

**Comment by kostmo** [comment]
## CircleCI build failures summary
As of commit 95e0653:
* **2/2** broken upstream at merge base 409151e (see [grid view](https://dr.pytorch.org/master-timeline.html?count=1&sha1=409151e1bb90c66546901d372f59930f72242ee9&should_suppress_scheduled_builds=true&should_suppress_fully_successful_columns=true&max_columns_suppress_successful=35))
    * You may want to rebase on the `viable/strict` branch (see its [recency history](https://dr.pytorch.org/master-viable-commits.html)):
        * If your commit is newer than `viable/strict`, you can try basing on an older, stable commit:
          ```
          git fetch viable/strict
          git rebase --onto viable/strict $(git merge-base origin/master HEAD)
          ```
        * If your commit is older than `viable/strict`:
          ```
          git fetch viable/strict
          git rebase viable/strict
          ```
* **0/2** failures introduced in this PR



## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=95e0653374c5dfcbd8f7951083a32aa6e729f536).

### 2 upstream failures recognized by patterns:
These builds matched patterns, but were probably caused by upstream breakages:
* [caffe2_onnx_py3_6_clang7_ubuntu16_04_test](https://dr.pytorch.org/build-details.html?build_id=75432095)
* [caffe2_onnx_py2_gcc5_ubuntu16_04_test](https://dr.pytorch.org/build-details.html?build_id=75432773)



---

<details><summary>This comment was automatically generated by <a href="https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment">Dr. CI</a> (expand for details).</summary>Follow <a href="https://dr.pytorch.org/admin/comments-opt-out.html">this link to opt-out</a> of these comments for your Pull Requests.

Please report bugs/suggestions on the <a href="https://github.com/kostmo/circleci-failure-tracker/issues">GitHub issue tracker</a>.
</details>

File: test/test_namedtensor.py [hunk 1]
```diff
@@ -272,6 +272,15 @@ def test_diagonal(self):
         self.assertEqual(named_tensor.diagonal(outdim='E', dim1='B', dim2='D').names,
                          ['A', 'C', 'E'])
 
+    def test_max_polling(self):
```
Comment (zou3519) [inline test/test_namedtensor.py]: s/max_polling/max_pooling/

**Review by zou3519 — APPROVED** [review]
lgtm, thank you!

Note that this doesn't provide complete support for F.max_pool1d: this PR implements support for F.max_pool{N}d with `return_indices=False`.

max_pool takes an optional `return_indices` argument; if `return_indices` is True, then it dispatches to `at::max_pool_1d_with_indices`, so it would be good to have named tensor support for the `at::max_pool_{N}d_with_indices` functions in the future.

**Comment by glaringlee** [comment]
Will have another version to support both max_pool and max_pool_with_indices

**Pull request closed without merging**